### PR TITLE
Ingore unit tests in mypy check

### PR DESCRIPTION
**Pull request by flash1293**
As discussed offline, this PR disables the CI mypy check for unit tests, as these are not as relevant to code stability as the main source code and currently contain a large number of violations.

Re-enabling this check is tracked here: https://github.com/airbytehq/airbyte-internal-issues/issues/1963

### CODE REVIEW

File: airbyte-cdk/python/bin/run-mypy-on-modified-files.sh [hunk 1]
```diff
@@ -1,2 +1,3 @@
 set -e
-git diff --name-only --relative --diff-filter=d remotes/origin/master -- . | grep -E '\.py$' | xargs .venv/bin/python -m mypy --config-file mypy.ini --install-types --non-interactive
\ No newline at end of file
+# TODO change this to incluyde unit_tests as well once it's in a good state
```
Comment (girarda) [inline airbyte-cdk/python/bin/run-mypy-on-modified-files.sh:2]:
```suggestion
# TODO change this to include unit_tests as well once it's in a good state
```

**Review by girarda — APPROVED** [review]

File: airbyte-cdk/python/bin/run-mypy-on-modified-files.sh [hunk 2]
```diff
@@ -1,2 +1,3 @@
 set -e
-git diff --name-only --relative --diff-filter=d remotes/origin/master -- . | grep -E '\.py$' | xargs .venv/bin/python -m mypy --config-file mypy.ini --install-types --non-interactive
\ No newline at end of file
+# TODO change this to include unit_tests as well once it's in a good state
+git diff --name-only --relative --diff-filter=d remotes/origin/master -- airbyte_cdk source_declarative_manifest | grep -E '\.py$' | xargs .venv/bin/python -m mypy --config-file mypy.ini --install-types --non-interactive
```
Comment (maxi297) [inline airbyte-cdk/python/bin/run-mypy-on-modified-files.sh:3]: We just need to make sure that whenever we will add a new package, we will have to update this. Can we work by exclusion instead? I see that `[tool.mypy] exclude` could do the trick

**Review by maxi297 — COMMENTED** [review]
Question regarding how we can make this easier to maintain

File: airbyte-cdk/python/bin/run-mypy-on-modified-files.sh (hunk 2 above)
Comment (flash1293) [reply airbyte-cdk/python/bin/run-mypy-on-modified-files.sh:3]: Fine with that too, feel free to update the PR @maxi297 - I can also do it tomorrow morning

Comment (flash1293) [reply airbyte-cdk/python/bin/run-mypy-on-modified-files.sh]: Switched it to an ignore list instead of an allow list on the git diff level

**Pull request merged**